src/components/ProgressBar/ProgressBar.tsx

```diff
@@ -0,0 +1,24 @@
+import { FC } from "react";
+
+import { Track, Progress } from "./ProgressBar.styles";
+
+const ProgressBar: FC<{
+  startTime: number;
+  endTime: number;
+}> = ({ startTime, endTime }) => {
+  const now = Date.now();
+  const done = now - startTime;
+  const left = endTime - startTime;
+  const fractionDone = done / left;
+  return (
+    <Track>
+      <Progress
+        animate={{ scaleX: 1 }}
+        initial={{ scaleX: fractionDone }}
+        transition={{ ease: "linear", duration: left / 1000 }}
+      ></Progress>
```

Comment:
Both this and the <span></span> below in src/components/TransactionsTab/subcomponents/WalletTransaction/WalletTransaction.tsx can be collapsed as their bodies are empty.